stdlib/gettext.pyi

```diff
@@ -27,7 +27,18 @@ class GNUTranslations(NullTranslations):
     CONTEXT: str
     VERSIONS: Sequence[int]
 
-def find(domain: str, localedir: StrPath | None = ..., languages: Iterable[str] | None = ..., all: bool = ...) -> Any: ...
+@overload
+def find(
+    domain: str, localedir: StrPath | None = ..., languages: Iterable[str] | None = ..., all: Literal[True] = ...
+) -> list[str]: ...
+@overload
+def find(
+    domain: str, localedir: StrPath | None = ..., languages: Iterable[str] | None = ..., all: Literal[False] = ...
+) -> str | None: ...
+@overload
+def find(
+    domain: str, localedir: StrPath | None = ..., languages: Iterable[str] | None = ..., all: bool = ...
+) -> list[str] | str | None: ...
```

Comment:
I feel like it might be better to continue returning `Any` for the fallback overload 🙂

```suggestion
) -> Any: ...
```

Reply:
Why? 🤔 

I think that three types are better than `Any`. They can be narrowed more easily.

Reply:
It conforms more to typeshed's general policy
(https://github.com/python/typeshed/blob/master/CONTRIBUTING.md, line 427 :)

Reply:
Ok!